RELEASE-PROCESS.md

```diff
@@ -0,0 +1,38 @@
+# Release Process
+
+External facing [Traffic Director](https://cloud.google.com/traffic-director)
+documentation uses the wallet example to showcase various xDS features. Here is
+one such [example][wallet-example-in-td-docs] where a specific branch of the
+wallet example is used.
+
+[wallet-example-in-td-docs]:
+https://cloud.google.com/traffic-director/docs/proxyless-configure-advanced-traffic-management#verify-routing-configuration
+
+# When to make a new version
+Whenever we have a new iteration of the public documentation, referencing the
+wallet example to showcase new set of functionality, we must make a new branch.
+Public documentation for the new set of functionality would end up referencing
+the newly created branch. This would ensure that existing documentation wouldn't
+need any changes, and any changes we make to the examples to support new
+functionality would not end up breaking old documentation inadvertently.
+
+# How to make a new version
+Once all changes are made to examples to support the new functionality, we
+should create a new branch. The public documentation will always refer to `HEAD`
+of these branches. This approach has the advantage that any bug fixes that need
+to be made on a branch do not require any update to the documentation.
+
+The branch corresponding to the first release version on this repo was v1.0.x.
+But going forward, we would just do v{VERSION}.x.
```

Comment:
Random thought: we could also name the branches with the name of the feature(s) being highlighted.

Reply:
What do you think about using `git branch --edit-description`?

Reply:
That could be helpful if we use versions and don't name the branches according to the feature.

The thing I was commenting about was that it's awkward to even use version numbers at all, since there isn't even really an implication of continuity here.  We could jettison the wallet client & service and use something completely different one day.  We could remove features from the demo.  The fact that one is called "v3" and one is called "v4" is entirely arbitrary.  But I also don't feel strongly about this, so if we want to use versions to avoid the need to come up with a name for the branch, I'm fine with that too.

Reply:
I'm not against the idea of naming the branches with feature names. But I'll go ahead and merge this one as is, and if we have enough interest to do that going forward, we can always do that. Thanks :)